packages/web-components/fast-foundation/src/tree-view/tree-view.ts

```diff
@@ -47,20 +57,36 @@ export class TreeView extends FoundationElement {
         });
     }
 
+    public currentFocused: HTMLElement | TreeItem | null;
+
     private treeItems: HTMLElement[];
 
     public handleBlur = (e: FocusEvent): void => {
         const { relatedTarget, target } = e;
+        if (
+            (relatedTarget === null || relatedTarget instanceof HTMLElement) &&
+            target instanceof HTMLElement &&
+            !this.contains(relatedTarget)
+        ) {
+            const treeView = this as HTMLElement;
+            treeView.setAttribute("tabindex", "0");
+        }
+    };
+
+    public handleFocus = (e: FocusEvent): void => {
+        const { relatedTarget, target } = e;
 
-        /**
-         * Clean up previously focused item's tabindex if we've moved to another item in the tree
-         */
         if (
-            relatedTarget instanceof HTMLElement &&
+            (relatedTarget === null || relatedTarget instanceof HTMLElement) &&
```

Comment:
why are you moving forward with the `.contains()` call if `relatedTarget` is null? Shouldn't this check for `!==` and skip the `.contains()` check if it is?